From 9f7a64279949023aaa396c96333e7444319c18fe Mon Sep 17 00:00:00 2001
From: Nithin Dabilpuram <ndabilpuram@marvell.com>
Date: Thu, 30 Aug 2018 13:00:36 +0530
Subject: [PATCH 0176/1051] soc: octeontx2: change usage of
 NIX_AF_TL3_TL2X_LINKX_CFG

Use TL2's index instead of TL3's index to reset
NIX_AF_TL3_TL2X_LINKX_CFG() as default value of
NIX_AF_PSE_CHANNEL_LEVEL is selecting TL2 and not TL3

Signed-off-by: Nithin Dabilpuram <ndabilpuram@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_nix.c | 6 +++---
 drivers/soc/marvell/octeontx2/rvu_reg.c | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_nix.c b/drivers/soc/marvell/octeontx2/rvu_nix.c
index 4215c4d12903..558b891021a9 100644
--- a/drivers/soc/marvell/octeontx2/rvu_nix.c
+++ b/drivers/soc/marvell/octeontx2/rvu_nix.c
@@ -964,10 +964,10 @@ static void nix_reset_tx_linkcfg(struct rvu *rvu, int blkaddr,
 	if (lvl == NIX_TXSCH_LVL_TL4)
 		rvu_write64(rvu, blkaddr, NIX_AF_TL4X_SDP_LINK_CFG(schq), 0x00);
 
-	if (lvl != NIX_TXSCH_LVL_TL3)
+	if (lvl != NIX_TXSCH_LVL_TL2)
 		return;
 
-	/* Reset TL3's CGX or LBK link config */
+	/* Reset TL2's CGX or LBK link config */
 	for (link = 0; link < (hw->cgx_links + hw->lbk_links); link++)
 		rvu_write64(rvu, blkaddr,
 			    NIX_AF_TL3_TL2X_LINKX_CFG(schq, link), 0x00);
@@ -1185,7 +1185,7 @@ static int nix_txschq_free(struct rvu *rvu, u16 pcifunc)
 	/* Disable TL2/3 queue links before SMQ flush*/
 	mutex_lock(&rvu->rsrc_lock);
 	for (lvl = NIX_TXSCH_LVL_TL4; lvl < NIX_TXSCH_LVL_CNT; lvl++) {
-		if ((lvl != NIX_TXSCH_LVL_TL3) && (lvl != NIX_TXSCH_LVL_TL4))
+		if ((lvl != NIX_TXSCH_LVL_TL2) && (lvl != NIX_TXSCH_LVL_TL4))
 			continue;
 
 		txsch = &nix_hw->txsch[lvl];
diff --git a/drivers/soc/marvell/octeontx2/rvu_reg.c b/drivers/soc/marvell/octeontx2/rvu_reg.c
index 5d0346c9e452..0c8622210f09 100644
--- a/drivers/soc/marvell/octeontx2/rvu_reg.c
+++ b/drivers/soc/marvell/octeontx2/rvu_reg.c
@@ -33,9 +33,9 @@ static struct hw_reg_map txsch_reg_map[NIX_TXSCH_LVL_CNT] = {
 	{NIX_TXSCH_LVL_SMQ, 2, 0xFFFF, {{0x0700, 0x0708}, {0x1400, 0x14C8} } },
 	{NIX_TXSCH_LVL_TL4, 3, 0xFFFF, {{0x0B00, 0x0B08}, {0x0B10, 0x0B18},
 			      {0x1200, 0x12E0} } },
-	{NIX_TXSCH_LVL_TL3, 4, 0xFFFF, {{0x1000, 0x10E0}, {0x1600, 0x1608},
-			      {0x1610, 0x1618}, {0x1700, 0x1768} } },
-	{NIX_TXSCH_LVL_TL2, 1, 0xFFFF, {{0x0E00, 0x0EE0} } },
+	{NIX_TXSCH_LVL_TL3, 3, 0xFFFF, {{0x1000, 0x10E0}, {0x1600, 0x1608},
+			      {0x1610, 0x1618} } },
+	{NIX_TXSCH_LVL_TL2, 2, 0xFFFF, {{0x0E00, 0x0EE0}, {0x1700, 0x1768} } },
 	{NIX_TXSCH_LVL_TL1, 1, 0xFFFF, {{0x0C00, 0x0D98} } },
 };
 
-- 
2.17.1

